x-pack/plugins/endpoint/public/applications/endpoint/store/policy_details/middleware.ts

```diff
@@ -17,25 +23,78 @@ export const policyDetailsMiddlewareFactory: MiddlewareFactory<PolicyDetailsStat
 
     if (action.type === 'userChangedUrl' && isOnPolicyDetailsPage(state)) {
       const id = policyIdFromParams(state);
+      let policyItem: PolicyData;
 
-      const { item: policyItem } = await sendGetDatasource(http, id);
+      try {
+        policyItem = (await sendGetDatasource(http, id)).item;
+      } catch (error) {
+        dispatch({
+          type: 'serverFailedToReturnPolicyDetailsData',
+          payload: error.body || error,
+        });
+        return;
+      }
+
+      // FIXME: remove this code once the Default Policy is available in the endpoint package - see: https://github.com/elastic/endpoint-app-team/issues/295
+      // Until we get the Default configuration into the Endpoint package so that the datasource has
+      // the expected data structure, we will add it here manually.
+      if (!policyItem.inputs.length) {
```

Comment:
@kevinlog - this is where we are injecting the "default" policy into the returned data from the API if not already there (which right now will always be the case)